commons/zenoh-shm/src/api/provider/types.rs

```diff
@@ -61,14 +59,18 @@ impl Default for AllocAlignment {
 
 impl AllocAlignment {
     #[zenoh_macros::unstable_doc]
-    pub fn new(pow: u8) -> Self {
-        Self { pow }
+    pub fn new(pow: u8) -> Result<Self, ZLayoutError> {
+        match pow {
+            pow if pow < 64 => Ok(Self { pow }),
+            _ => Err(ZLayoutError::IncorrectLayoutArgs),
+        }
     }
 
     /// Get alignment in normal units (bytes)
     #[zenoh_macros::unstable_doc]
-    pub fn get_alignment_value(&self) -> usize {
-        1usize << self.pow
+    pub fn get_alignment_value(&self) -> NonZeroUsize {
+        // SAFETY: this is safe
+        unsafe { NonZeroUsize::new_unchecked(1usize << self.pow) }
```

Comment:
`AllocAlignment::new()` takes `pow: u8` that returns `Ok` if `pow < 64`.
`1usize << self.pow` may overflow in case `usize` is `32bit`.

Reply:
Fixed through `usize::BITS` as @wyfo suggested

Reply:
@wyfo  my understanding is:
1. Fist of all, `1usize << pow` is evaluated, and it's result is `usize`. An overflow might happen here, which will panic on Debug and be UB on Release (UB because it is processor-defined)
2. In `NonZeroUsize::new` compiler is able to guess that ON TARGET ARCHITECTURE any shift operation applied to `1usize` constant at step 1 will give nonzero result, even in case of overflow. It means that for TARGET ARCH a check may be elided, but fir other architectures this might be wrong and check assembly will be generated.